src/client/components/Meals.js

```diff
@@ -0,0 +1,135 @@
+import React, { useState, useEffect } from "react";
+import Meal from "./Meal";
+import images from "./images";
+import common from "../assets/images/common.png";
+
+import { Route, Switch, Link, useRouteMatch } from "react-router-dom";
+
+const Meals = () => {
+  const [meals, setMeals] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+  const [query, setQuery] = useState("");
+  let { path, url } = useRouteMatch();
+
+  useEffect(() => {
+    fetchMealsApi();
+  }, []);
+
+  const fetchMealsApi = async () => {
```

Comment:
Try and see if your can reuse your fething and loading and error handling in some way.